src/components/structures/auth/Login.js

```diff
@@ -339,6 +344,20 @@ export default createReactClass({
         this.props.onRegisterClick();
     },
 
+    onTryRegisterClick: function(ev) {
+        if (this._isSsoStep()) {
```

Comment:
there's something I don't like about calling `_getCurrentFlowStep` three times for one click of a button

Reply:
Yea, it made more sense when we were using it in a couple places on the release branch. I've internalized it now.